src/main/java/net/minestom/server/adventure/provider/TerminalColorConverter.java

```diff
@@ -0,0 +1,108 @@
+package net.minestom.server.adventure.provider;
+
+import net.kyori.adventure.text.format.NamedTextColor;
+import net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer;
+import net.minestom.server.utils.PropertyUtils;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * A string converter to convert a string to an ansi-colored one.
+ *
+ * @see <a href="https://github.com/Minecrell/TerminalConsoleAppender/blob/master/src/main/java/net/minecrell/terminalconsole/MinecraftFormattingConverter.java">TerminalConsoleAppender</a>
+ * @see <a href="https://github.com/PaperMC/Paper/blob/41647af74caed955c1fd5b38d458ee59298ae5d4/patches/server/0591-Add-support-for-hex-color-codes-in-console.patch">Paper</a>
+ */
+public final class TerminalColorConverter {
```

Comment:
Should be tested.
Also do we really need to expose the class? Couldn't it become package-private in `terminal`?

Reply:
> Should be tested. Also do we really need to expose the class? Couldn't it become package-private in `terminal`?

Probably not. This may be useful if someone want to make their own terminal.

Reply:
I'd be for hiding it at least for now. And the terminal package seems more fitting anyway.

Reply:
> I'd be for hiding it at least for now. And the terminal package seems more fitting anyway.

Not just terminal use this. `MinestomComponentLoggerProvider` need this too, at least the `SERIALIZER` of the converter. I just don't want to duplicate `COLOR_CHAR` just for the package-private one.
Also, if a developer needs this for their own terminal, they don't have to make an PR to change it back.

Reply:
Or I'm just bad at creating classes that split the purpose for both `MinestomComponentLoggerProvider` and the `terminal`.

Reply:
but I'll move the converter to `terminal` anyways

Reply:
I am indeed not sure that the serializer has to be here

Reply:
I take this back. `MinestomFlattenerProvider` is used in the converter. Should I public that instance @TheMode ?

Reply:
Cant you move the serializer in `MinestomComponentLoggerProvider`? `COLOR_CHAR` being duplicated doesn't sound like a huge issue. I'd prefer limiting exposed stuff to the strict minimum until necessary

Reply:
> Cant you move the serializer in `MinestomComponentLoggerProvider`? `COLOR_CHAR` being duplicated doesn't sound like a huge issue. I'd prefer limiting exposed stuff to the strict minimum until necessary

I just got a better idea to use `SECTION_CHAR` from `LegacyComponentSerializer` instead of creating a constant for it.
Okay, I've moved the converter to the terminal package. Not sure about package-private thing, but I'd leave them as public for developers who want to create their own terminal.

Reply:
There is nothing we do there that third party cannot do, may as well hide it to allow ourselves some breaking change down the line